jni/src/native_peer.c

```diff
@@ -71,40 +71,62 @@ Java_tel_schich_libdatachannel_LibDataChannelNative_rtcCreatePeerConnection(JNIE
             .mtu = mtu,
             .maxMessageSize = maxMessageSize,
     };
+    
+    jstring* serverStrings = NULL;
+    
     if (iceServers != NULL) {
         config.iceServersCount = (*env)->GetArrayLength(env, iceServers);
         if (config.iceServersCount > 0) {
             config.iceServers = malloc(sizeof(char*) * config.iceServersCount);
-            if (config.iceServers == NULL) {
+            serverStrings = malloc(sizeof(jstring) * config.iceServersCount);
+            if (config.iceServers == NULL || serverStrings == NULL) {
+                free(config.iceServers);
+                free(serverStrings);
                 throw_native_exception(env, "Failed to allocate for ice servers!");
                 return EXCEPTION_THROWN;
             }
 
             for (int i = 0; i < config.iceServersCount; i++) {
-                config.iceServers[i] = (*env)->GetStringUTFChars(env, (*env)->GetObjectArrayElement(env, iceServers, i), NULL);
+                serverStrings[i] = (*env)->GetObjectArrayElement(env, iceServers, i); // we need a reference to release later
+                config.iceServers[i] = (*env)->GetStringUTFChars(env, serverStrings[i], NULL);
+                if (config.iceServers[i] == NULL) {
+                    // release everything and throw an exception
+                    for (int j = 0; j < i; j++) {
+                        (*env)->ReleaseStringUTFChars(env, serverStrings[j], config.iceServers[j]);
+                        (*env)->DeleteLocalRef(env, serverStrings[j]);
```

Comment:
my understand was, that local refs are freed automatically once the JNI call terminates.

Reply:
You are right, i reverted this part, but kept the `serverStrings` array to be used for `ReleaseStringUTFChars`, because i am not entirely sure that expecting `GetObjectArrayElement` to return exactly the same object twice is compliant with the specs. But i couldn't find a definitive answer.